### PR TITLE
Android Gradle Plugin and Gradle Locks update

### PR DESCRIPTION
**Pull request by Milan-Cerovsky**
> **Note**
>This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [ ] Update **documentation** as appropriate (e.g [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), [**CHANGELOG.md**](../blob/main/CHANGELOG.md), etc.)
- [x] **Run the app** and try the changes
- [x] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

> **Note**
> It is good practice to provide before and after UI  **screenshots** in the description of this PR. This is only applicable for changes that modify the UI.

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [x] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [x] Review the **documentation**, [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), etc. as appropriate
- [x] **Run the app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: buildSrc/gradle.lockfile [hunk 1]
```diff
@@ -1,33 +1,28 @@
 # This is a Gradle generated file for dependency locking.
 # Manual edits can break the build and are not advised.
 # This file is expected to be part of source control.
-com.googlecode.javaewah:JavaEWAH:1.1.13=buildScriptClasspath,compileClasspath,testCompileClasspath,testRuntimeClasspath
```
Comment (Milan-Cerovsky) [inline buildSrc/gradle.lockfile:4]: this file needed update, for some reasone we had some of version 1.7.* while others with 1.9.*

File: gradle.properties [hunk 2]
```diff
@@ -138,7 +138,7 @@ ANDROID_COMPILE_SDK_VERSION=34
 
 ANDROID_NDK_VERSION=23.0.7599858
 
-ANDROID_GRADLE_PLUGIN_VERSION=8.3.0
+ANDROID_GRADLE_PLUGIN_VERSION=8.5.0
```
Comment (Milan-Cerovsky) [inline gradle.properties:141]: minor change - bumped AGP. There were security and performance fixes. On my device the performance  improved dramatically

File: settings.gradle.kts [hunk 3]
```diff
@@ -54,17 +54,17 @@ pluginManagement {
...
-        kotlin("plugin.serialization") version (kotlinVersion)
+        id("com.android.application") version (androidGradlePluginVersion) apply false
+        id("com.android.library") version (androidGradlePluginVersion) apply false
+        id("com.android.test") version (androidGradlePluginVersion) apply false
+        id("com.github.ben-manes.versions") version (extra["GRADLE_VERSIONS_PLUGIN_VERSION"].toString()) apply false
+        id("com.osacky.fulladle") version (extra["FULLADLE_VERSION"].toString()) apply false
+        id("org.jetbrains.kotlinx.kover") version (extra["KOVER_VERSION"].toString()) apply false
+        id("wtf.emulator.gradle") version (extra["EMULATOR_WTF_GRADLE_PLUGIN_VERSION"].toString()) apply false
+        kotlin("android") version (kotlinVersion) apply false
+        kotlin("jvm") version (kotlinVersion) apply false
```
Comment (Milan-Cerovsky) [inline settings.gradle.kts:65]: I think applying all these plugins to each repo was unnecessary, having them in classpath and adding the explicitly like we always do is enough

Comment (HonzaR) [reply settings.gradle.kts:65]: Good point.

File: docs/Setup.md [hunk 4]
```diff
@@ -71,6 +71,9 @@ Start by making sure the command line with Gradle works first, because **all the
 1. Reboot your computer, which will ensure that Gradle and Kotlin daemons are completely killed and relaunched
 1. Delete the global Gradle cache under `~/.gradle/caches`
 1. If adding a new dependency or updating a dependency, a warning that a dependency cannot be found may indicate the Maven repository restrictions need adjusting
+1. If included build (like buildSrc) contains different kotlin version (embedded from gradle) than other modules
```
Comment (HonzaR) [inline docs/Setup.md:74]: We could also link the follow-up issue here in the known issues. Thank you for taking care of it.

**Review by HonzaR — COMMENTED** [review]
These changes are working well. It waits just for the follow-up, as commented inline.

**Review by HonzaR — APPROVED** [review]
Thanks for these changes @Milan-Cerovsky. It's ready to go now!

**Pull request merged**